lib/utf8_sanitizer.rb

```diff
@@ -42,12 +40,17 @@ def event_attributes(env, request)
       user_agent: request.user_agent,
       timestamp: Time.zone.now,
       host: request.host,
-      visitor_id: request.cookies['ahoy_visitor'],
+      visitor_id: sanitized_visitor_id(request),
       content_type: env['CONTENT_TYPE'],
     }.to_json
   end
 
   def remote_ip(request)
     @remote_ip ||= (request.env['action_dispatch.remote_ip'] || request.ip).to_s
   end
+
+  def sanitized_visitor_id(request)
+    request.cookies['ahoy_visitor']&.
+      encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
```

Comment:
This logic is duplicated in the ahoy initializer, would be make sense to extract it to a single place?